src/regularized/regularized.jl

```diff
@@ -0,0 +1,46 @@
+"""
+optimizer: θ ⟼ argmax θᵀy - Ω(y)
+"""
+struct Regularized{O,R}
+    Ω::R
+    optimizer::O
```

Comment:
I would rather call this `concave_maximizer` to differentiate from `(linear_)maximizer` used elsewhere